### ISSUE TITLE
`acceptLiteralFieldnames(true)` no longer seems to be honored

### ISSUE THREAD

**Issue by rdesgroppes** [HubSpot/jackson-datatype-protobuf#102]
It's very likely that #65 led `acceptLiteralFieldnames(true)` to become ineffective.

**Comment by jhaber** [comment]
Do you have an example of the behavior you're seeing, and what behavior you expected? I just added a test to make sure that messages with json_name still accept the literal field names and it seems to work fine:
https://github.com/HubSpot/jackson-datatype-protobuf/compare/jh/json-names-test?expand=1

**Comment by rdesgroppes** [comment]
In `PropertyNamingCache`, it turns out that `field.toProto().hasJsonName()` always return `true` in my case, leading to `tempMap`s defined in `buildSerializationFunction` and `buildDeserializationFunction` to disregard the configured `namingStrategy` (passed to `ObjectMapper`).
Locally undoing changes #82 gets both features work again.
Now trying to isolate reproducible steps...

**Comment by rdesgroppes** [comment]
> it turns out that `field.toProto().hasJsonName()` always return `true` in my case

Well, not only in my case: it turns out `protoc` _alway_s populates `json_name`: https://github.com/protocolbuffers/protobuf/issues/5587

**Comment by jhaber** [comment]
Interesting, what version of protoc/protobuf-java are you using?

**Comment by rdesgroppes** [comment]
> Interesting, what version of protoc/protobuf-java are you using?

- `com.google.protobuf:protoc:exe:*:3.21.12`
- `com.google.protobuf:protobuf-java:3.21.12`

**Comment by jhaber** [comment]
Hmm I'm still not having any luck reproducing, even with that version:
https://github.com/HubSpot/jackson-datatype-protobuf/compare/jh/json-names-test?expand=1

Is it possible the protobuf was compiled with an older protoc?

**Comment by rdesgroppes** [comment]
> Is it possible the protobuf was compiled with an older protoc?

No, I double-checked the whole toolchain (even replacing "my" `protoc.exe` to make sure no magically bundled one is used).
For each field like:
```proto
    int32 Int32_With_Different_Field_Name = 2;
```

One may see in the generated sources things like:
```java
    /**
     * <code>int32 Int32_With_Different_Field_Name = 2 [json_name = "Int32WithDifferentFieldName"];</code>
     * @return The int32WithDifferentFieldName.
     */
    int getInt32WithDifferentFieldName();
```

Would you be so kind as to give a try with such an unusual field name, i.e. `Int32_With_Different_Field_Name`?

**Comment by jhaber** [comment]
Yeah, I'm not seeing that in the generated code with protoc 3.6.0 and 3.21.12:
```java
    /**
     * <code>int32 Int32_With_Different_Field_Name = 2;</code>
     * @return The int32WithDifferentFieldName.
     */
    int getInt32WithDifferentFieldName();
```

**Comment by rdesgroppes** [comment]
Right away, please hold the line. And thank you for your efforts!

**Comment by rdesgroppes** [comment]
... works!

**Comment by jhaber** [comment]
👍 thanks for testing, I'll open a PR. And I'll keep playing around to see if I can get that json_name attribute added by default so we can have a regression test

**Comment by rdesgroppes** [comment]
Thank _you_ for taking care!

What happens, if I understand correctly:
1. when `protoc` is invoked as a **standalone** tool, it doesn't add `json_name`s by itself (but honors user-defined ones),
2. when `protoc` is invoked as a **driver** for a plugin (_a la_ `gcc`), then it _does add_ `json_name`s "in case of", which happens in my case - for which an issue got filed: https://github.com/protocolbuffers/protobuf/issues/5587.

**Comment by rdesgroppes** [comment]
Actually not, the right discrepancy is that:
> protoc: code gen is different between using source files and using descriptor set

... which is perfectly well described by https://github.com/protocolbuffers/protobuf/issues/6175.
